src/contexts/ordersContext.tsx

```diff
@@ -184,15 +207,21 @@ function getOrderStatus({
   }
 }
 
-export async function getSingleOrder(
-  orderHash: Address,
-  userAddress: Address,
-  publicClient: PublicClient
-) {
-  return publicClient.readContract({
-    address: COMPOSABLE_COW_ADDRESS,
-    abi: composableCowAbi,
-    functionName: "singleOrders",
-    args: [userAddress, orderHash],
-  });
+  return (
+    <OrderContext.Provider
+      value={{
+        orders,
+        loaded,
+        error,
+        reload,
+      }}
+    >
+      {children}
+    </OrderContext.Provider>
+  );
+}
+
+export function useOrder() {
+  const context = useContext(OrderContext);
+  return context;
 }
```

Comment:
did this end up making any difference in the UI?

Reply:
now we don't need to fetch all the order data again when going make and forth on the app. It only calculates on the first render